### DISCUSSION TITLE
Adding support for EV-chargers ?

### DISCUSSION THREAD

**Discussion by oscarhemberg** [zbigniewmotyka/home-assistant-solplanet#54] [Ideas]
Hi,
I have a large Solplanet installation and just found your integration. Installed perfectly and found both my inverters and both battery stacks and everything seems to run well. Just wanted to say thank you.

If you ever decide to add support also for their EV-chargers let me know. I have 2 of those as well and would be happy to help test.

Keep up the good work.

**Comment by zubir2k** [comment]
I am also curious which code represent EV charger. Perhaps could you share the output from this api:
http://`inverter ip`:8484/getdev.cgi?device=`code`

Code that we know so far:
1 = wifi dongle
2 = inverter
3 = smart meter
4 = battery

you could try with `0` or `5` and see if there's any response

**Comment by oscarhemberg** [comment]
I get the same response with 0 and 5,6,7,... but they are just the dongle
and same response. I however get no response with 1.

The EV chargers have separate IPs and connected to cloud directly I believe
and not through dongle. I tried using the call with the EV charger IP but
it refuses to connect.

**Comment by zbigniewmotyka** [comment]
@oscarhemberg, how you can configure this ev charger (f.ex. to specify the WiFi network)? During this configuration, the solplanet application connects to the charger using WiFi? If yes, then it must be possible to connect somehow to this charger. If it is configured using another method (f.ex. Bluetooth ble) then there is a risk that it could be hard to read data in a similar way as for inverter.

**Comment by oscarhemberg** [comment]
They support BLE, wifi or Ethernet (and I think also RS585 but that I have
not tried). I have mine with ethernet cable and they have separate IPs just
through normal DHCP. Setup in the Solplanet App by scanning the QR code
same as for the inverters and batteries. So for sure possible to connect to
them but I just dont know what port and command they listen to.

--Oscar

**Comment by zubir2k** [comment]
@oscarhemberg if the charger has indeed linked to the Solplanet app, then the local API should return with the EV charger values. Perhaps you could try login to the web (https://internation-cloud.solplanet.net/user/login) and see if you're able to view the EV charger there.. 

![image](https://github.com/user-attachments/assets/b8699ccd-b999-483c-954d-800bad6dc611)

**Comment by oscarhemberg** [comment]
Yes, I see them there.

//Oscar

**Comment by zbigniewmotyka** [comment]
A good point to start would be to use nmap (https://nmap.org/) to check if the charger has any open ports by running command `nmap -vv -p- <charger-ip-address>` (f.ex. `nmap -vv -p- 192.168.1.100`).

**Comment by zubir2k** [comment]
> I get the same response with 0 and 5,6,7,... but they are just the dongle and same response. I however get no response with 1. The EV chargers have separate IPs and connected to cloud directly I believe and not through dongle. I tried using the call with the EV charger IP but it refuses to connect.

@oscarhemberg Come to think of it, the response that you got for device code = 0, 5-7 is pointing to the wifi dongle instead of code 1. In my case, I got 404 response if I tried calling device code 0, 5, 6 and 7. Perhaps maybe if you could share the json that you got earlier for these codes.  Logically, if connected OK in Solplanet, the data should be available.

@zbigniewmotyka imho, its best that we stick to 1 source of connection. Unless of course the Solplanet logger doesn't include EV charger to begin with 😉

**Comment by oscarhemberg** [comment]
Attached are the JSON files. for code=1 I just get "request not found, 404"
reply. and for 6,7,8... It is the same as 5. WHen I look at them I get
confused as I can only see the response of one battery stack. I have two
inverters in parallel and each has a separate battery stack. In the App/web
everything shows up correctly including the smart meter and 2x battery
packs. Also in HA the batteries both show up although I can not see one of
them in the JSON files. I also tried the Nmap but could find no open
ports on one of the EVchargers.

--Oscar

[image: Screenshot 2024-11-12 074546.png]

**Comment by zubir2k** [comment]
> Attached are the JSON files. for code=1 I just get "request not found, 404" reply. and for 6,7,8... It is the same as 5. WHen I look at them I get confused as I can only see the response of one battery stack. I have two inverters in parallel and each has a separate battery stack. In the App/web everything shows up correctly including the smart meter and 2x battery packs. Also in HA the batteries both show up although I can not see one of them in the JSON files. I also tried the Nmap but could find no open ports on one of the EVchargers.

Hi Oscar @oscarhemberg, your response earlier doesnt have the JSON file. could you resend it again here. Thanks

**Comment by oscarhemberg** [comment]
Strange, here they are again.

--Oscar

**Comment by Tjepp** [comment]
Hi! I also have a Solplanet charger and would love this integration. Any way I can help? 😄

**Comment by zubir2k** [comment]
Hi @Tjepp 
Could you help to check what response did you get when accessing to this API?
Try with code `0` or `5` and see if there's any response.

The API: http://`inverter ip`:8484/getdev.cgi?device=`code`

Code that we know so far: 
1 = wifi dongle 
2 = inverter 
3 = smart meter 
4 = battery

![image](https://github.com/user-attachments/assets/cb7277d4-2d6f-42f8-ab9c-c11ae476b0bd)

**Comment by Tjepp** [comment]
Both 0 and 5 replies "request not found, 404"

**Comment by M3nt0san** [comment]
Hello, 
Did anyone manage to use Solplanet chargers with the integration? I can also help with some testing if there's someone willing to help out :).
Except for this fact, the integration works fasntastic for me, Kudos @zbigniewmotyka.

**Comment by zbigniewmotyka** [comment]
@M3nt0san Thanks! Glad to hear that — appreciate the feedback!

Right now, there is no known way to get data from the EV charger. If you find a way to read these values, some instructions would allow me to try adding support.

**Comment by nielshojen** [comment]
I got my charger installed today, and using device=5 I now get:

{"diesel_power":0,"diesel_status":0,"diesel_mode":0,"diesel_vac_phs":[0,0,0],"diesel_iac_phs":[0,0,0],"diesel_pac_phs":[0,0,0],"diesel_fac":0}

I haven't had a car plugged in to it yet, but I'll get back with an update when my wife get's back with the car :)

**Comment by Ir0nyPT** [comment]
I'm currently facing a similar issue, I'm trying to get the values from my second energy meter that is measuring the AC coupling but I'm finding no API for that 

![IMG_20250820_162004](https://github.com/user-attachments/assets/ae81599d-1209-46ed-b9bb-184e2421583f)


http://Ip:8484/getdev.cgi?device=3

{
  "mod": 2,
  "enb": 1,
  "exp_m": 0,
  "regulate": 10,
  "enb_PF": 0,
  "target_PF": 0,
  "total_pac": 0,
  "total_fac": 3680,
  "meter_pac": 515,
  "sn": "EASTRONB3240H3302B4",
  "manufactory": "EASTRON",
  "type": "1",
  "name": "SDM230",
  "model": 2,
  "abs": 0,
  "abs_offset": 0,
  "sec_enb": 1,
  "sec_sn": "CHNTB3240H3302B4SUB",
  "sec_manufactory": "CHNT",
  "sec_name": "DDSU666",
  "sec_mod": 21,
  "sec_meter_location": 0,
  "sec_addr": 2
}


I'm trying to sniff the packages over the lan but I'm still trying to learn how to do it ;)

**Comment by Ir0nyPT** [comment]
I think this is some increment
<img width="396" height="49" alt="image" src="https://github.com/user-attachments/assets/d9337a5b-488d-4d10-b329-76af2add4cf9" />


How did I get this?
I installed "PCAPdroid" and recorded to a pcap, then I open the app as an Installer and navigated on the app.
After that sent to the computer and opened on wireshark.

I think you could do the same to check if there is any url with more info

**Comment by josujs** [comment]
Hello, first of all, my sincerest congratulations on the integration; it's fantastic, thank you very much.
I also have a Solplanet electric charger installed. If I could help you, let me know. 
The only thing is that my computer skills are very limited. Thank you very much for all your time and work.

Regards
